### PR TITLE
Feat/add users sitegroups tables

### LINKED ISSUE #73: Multiple Clients
We want to upgrade the datamodel so we can have multiple clients viewing different sites. 

## Detailed Description
Currently, we have one client, that has all the sites attached to it. 

We want to be able to do the following
1. one user - can add or view one site
2. one user, can add or view multiple sites
3. Two users (for example from the sample company), want to look at one site
4. Two users, wanting to look at multiple sites (could be added by another user). Any user from site group can add a site. 
5. OCF user want to see everything (admin)

Datamodel proposal:

```mermaid
  graph TD;
      User-- N:1 -->SiteGroup;
      SiteGroup-- N:N -->Site;
```
- One `user` is in one `sitegroup`. Each site group can have multiple users. 
- Each `sitegroup` contains multiple `sites`. One `site` can be in multiple `sitegroups`

1. one user - one site

```mermaid
  graph TD;
      A(User=Alice)-->B(SiteGroup=Alice1);
      B --> C(Site);
```

2. one user - two sites

```mermaid
  graph TD;
      A(User=Alice)-->B(SiteGroup=Alice1);
      B --> C1(Site1);
B --> C2(Site2);
```

3. Two users - one site

```mermaid
  graph TD;
      A1(User=Alice)-->B(SiteGroup);
A2(User=Bob)-->B(SiteGroup);
      B --> C1(Site1);
```

4. Two users - two site

```mermaid
  graph TD;
      A1(User=Alice)-->B(SiteGroup);
A2(User=Bob)-->B(SiteGroup);
      B --> C1(Site1);
B --> C2(Site2);
```

5. OCF can see everything

```mermaid
  graph TD;
      A1(User=Alice)-->B(SiteGroup1);
A2(User=Bob)-->B(SiteGroup1);
A3(User=OCF)-->B2(SiteGroup2);
      B --> C1(Site1);
B --> C2(Site2);
      B2 --> C1(Site1);
B2 --> C2(Site2);
B2 --> C3(Site3);
```

## Context
- Soon we will have more than one customer on the site project
- There maybe some `public` sites, but we can assign them a `sitegroup`

## Possible Implementation
 - [x] Remove `Client` table in code
 - [x] Add `SiteGroup` table
 - [x] Add 'User' table, - see example [here](https://github.com/openclimatefix/nowcasting_datamodel/blob/main/nowcasting_datamodel/models/api.py#L28)
 - [x] Create `User` to `SiteGroup` relationship similar to [this](https://github.com/openclimatefix/pv-site-datamodel/blob/main/pvsite_datamodel/sqlmodels.py#L254)
 - [x] Create `Site` to `SiteGroup` relationship similar to [this](https://stackoverflow.com/questions/5756559/how-to-build-many-to-many-relations-using-sqlalchemy-a-good-example)
 - [x] Add migration for this change
 - [x] Add test that go through 1. to 5. above
 - [x] Update API get sites read function - https://github.com/openclimatefix/pv-site-api/pull/99
 - [x] Update API to add user if not in db, also make dummy site group - https://github.com/openclimatefix/pv-site-api/pull/99
 - [x] Upgrade forecasts and db clean up - https://github.com/openclimatefix/pv-site-production/pull/66
 - [x] Run migrations - remove client data
 - [x] Make sure OCF can see all sites
 - [x] Add tools on Analaysis dashboard too - add site to site group, add user to site group - https://github.com/openclimatefix/uk-analysis-dashboard/issues/39
 - [x] Upgrade PVConsumer - https://github.com/openclimatefix/PVConsumer/pull/70

dev
- [x] Upgrade datamodel db to 0.1.35
- [x] Upgrade api to 1.0.0
- [x] Upgrade forecasts 1.0.0
- [x] PVconsumer 1.0.0

pro
- [x] Upgrade datamodel db to 0.1.35
- [x] Upgrade api to 1.0.0
- [x] Upgrade forecasts 1.0.0
- [x] PVconsumer 1.0.0

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Upgrade datamodel for users, site and site groups
- removed client table
- added user table
- added sitegroup table
- added relations for user, sitegroup, and sites + tests
- added 'get_user_by_email' + test. This will be used by the API
- added sql migration

Fixes #73 

## How Has This Been Tested?

- added tests

- [x] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvsite_datamodel/read/generation.py [hunk 1]
```diff
@@ -6,29 +6,34 @@
...
+    user_uuids: Optional[List[str]] = None,
 ) -> List[GenerationSQL]:
-    """Get the generation data by client.
+    """Get the generation data by user uuids.
 
     :param session: database session
     :param start_utc: search filters >= on 'datetime_utc'. Can be None
     :param end_utc: search filters < on 'datetime_utc'. Can be None
-    :param client_names: optional list of provider names
+    :param user_uuids: optional list of provider names
```
Comment (peterdudfield) [inline pvsite_datamodel/read/generation.py:31]: update

File: tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    site_group.sites.append(site_1)
+    site_group.sites.append(site_2)
+
+    assert len(user_1.site_group.sites) == 2
+    assert len(user_2.site_group.sites) == 2
+
+
+def test_two_site_two_user_and_ocf_see_everything(db_session):
+    """
+    4. Test that one site with one user works
```
Comment (peterdudfield) [inline tests/test_models.py:74]: update

File: tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    # add site to site group
+    site_group.sites.append(site_1)
+
+    assert len(user_1.site_group.sites) == 1
+    assert len(user_2.site_group.sites) == 1
+
+
+def test_two_site_two_user(db_session):
+    """
+    4. Test that one site with one user works
```
Comment (peterdudfield) [inline tests/test_models.py:55]: update

**Review by devsjc — APPROVED** [review]
LGTM

**Pull request merged**